### PR TITLE
Adding validation for entity proxy interface

### PR DESCRIPTION
**Pull request by shibayan**
Hi @cgillum 

Added validation of interface used as Entity Proxy.
There was a problem that the error message was difficult to understand, so adding the problematic method and interface name made it easy to identify the cause.

Ref #859

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs [hunk 1]
```diff
@@ -47,6 +49,24 @@ private static Type CreateProxyType(Type interfaceType)
...
+            }
+
+            if (interfaceType.GetProperties(BindingFlags.Instance | BindingFlags.Public).Length > 0)
+            {
+                throw new InvalidOperationException($"Interface '{interfaceType.FullName}' can not define properties.");
+            }
+
+            if (interfaceType.GetMethods(BindingFlags.Instance | BindingFlags.Public).Length == 0)
+            {
+                throw new InvalidOperationException($"Interface '{interfaceType.FullName}' has no defined method.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs]:
```suggestion
                throw new InvalidOperationException($"Interface '{interfaceType.FullName}' has no methods defined.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs [hunk 2]
```diff
@@ -47,6 +49,24 @@ private static Type CreateProxyType(Type interfaceType)
...
+        private static void ValidateInterface(Type interfaceType)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new InvalidOperationException($"{interfaceType.Name} is not an interface.");
+            }
+
+            if (interfaceType.GetProperties(BindingFlags.Instance | BindingFlags.Public).Length > 0)
+            {
+                throw new InvalidOperationException($"Interface '{interfaceType.FullName}' can not define properties.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs]:
```suggestion
                throw new InvalidOperationException($"Interface '{interfaceType.FullName}' defines properties. Entity proxy interfaces with properties are not supported.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs [hunk 3]
```diff
@@ -83,15 +103,15 @@ private static void BuildMethods(TypeBuilder typeBuilder, Type interfaceType)
                 // check that the number of arguments is zero or one
                 if (parameters.Length > 1)
                 {
-                    throw new InvalidOperationException("Only a single argument can be used for operation input.");
+                    throw new InvalidOperationException($"Method '{methodInfo.Name}' is only a single argument can be used for operation input.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs]:
```suggestion
                    throw new InvalidOperationException($"Method '{methodInfo.Name}' defines more than one parameter. Entity proxy interface methods must define at most one argument for operation input.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs [hunk 4]
```diff
@@ -83,15 +103,15 @@ private static void BuildMethods(TypeBuilder typeBuilder, Type interfaceType)
...
                 }
 
                 var returnType = methodInfo.ReturnType;
 
                 // check that return type is void / Task / Task<T>.
                 if (returnType != typeof(void) && !(returnType == typeof(Task) || returnType.BaseType == typeof(Task)))
                 {
-                    throw new InvalidOperationException("Only a return type is void / Task / Task<T>.");
+                    throw new InvalidOperationException($"Method '{methodInfo.Name}' is only a return type is void / Task / Task<T>.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs]:
```suggestion
                    throw new InvalidOperationException($"Method '{methodInfo.Name}' has a return type which is neither void nor a Task. Entity proxy interface methods may only return void, Task, or Task<T>.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/DurableEntityProxyExtensions.cs [hunk 5]
```diff
@@ -115,12 +115,12 @@ private static Type CreateTypeMapping(Type interfaceType)
 
             if (!implementedTypes.Any())
             {
-                throw new InvalidOperationException($"Cannot found {interfaceType.Name} implemented type.");
+                throw new InvalidOperationException($"Cannot find class that implements {interfaceType.FullName}.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/DurableEntityProxyExtensions.cs]:
```suggestion
                throw new InvalidOperationException($"Could not find any types that implement {interfaceType.FullName}.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/DurableEntityProxyExtensions.cs [hunk 6]
```diff
@@ -115,12 +115,12 @@ private static Type CreateTypeMapping(Type interfaceType)
...
             }
 
             if (implementedTypes.Length > 1)
             {
-                throw new InvalidOperationException("Entity type is ambiguous.");
+                throw new InvalidOperationException($"Ambiguous derived class with implemented {interfaceType.FullName}.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/DurableEntityProxyExtensions.cs]:
```suggestion
                throw new InvalidOperationException($"Found multiple types that implement {interfaceType.FullName}. Only one type is allowed to implement an interface used for entity proxy creation.");
```

File: src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs [hunk 7]
```diff
@@ -47,6 +49,24 @@ private static Type CreateProxyType(Type interfaceType)
             return typeBuilder.CreateTypeInfo();
         }
 
+        private static void ValidateInterface(Type interfaceType)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new InvalidOperationException($"{interfaceType.Name} is not an interface.");
```
Comment (cgillum) [inline src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs]:
```suggestion
                throw new InvalidOperationException($"{interfaceType.Name} is not an interface. Entity proxy type parameters must be interfaces.");
```

**Comment by shibayan** [comment]
@cgillum Thank you for suggestions! It has been applied.

**Pull request merged**